cli/internal/login/login.go

```diff
@@ -61,12 +63,14 @@ func (c *LoginCommand) Run(args []string) int {
 		Short: "Login to your Vercel account",
 		RunE: func(cmd *cobra.Command, args []string) error {
 			login := login{
-				ui:                  c.UI,
-				logger:              c.Config.Logger,
-				openURL:             browser.OpenBrowser,
-				client:              c.Config.ApiClient,
-				writeUserConfig:     config.WriteUserConfigFile,
-				writeRepoConfig:     config.WriteRepoConfigFile,
+				ui:       c.UI,
+				logger:   c.Config.Logger,
+				fsys:     c.Config.Fs,
+				repoRoot: c.Config.Cwd,
```

Comment:
That's a super-confusing remapping. We should probably not call it `cwd` if it's actually the repo root.

Reply:
Heh. It's actually more confusing than it looks. We have potentially 3 paths in play:
 1. The actual working directory of the `turbo` process
 2. The directory that contains `turbo.json` -> the "root" of the monorepo
 3. The directory that contains the `.git` folder
 
 None of them are guaranteed to be the same. We most often care about the second in the list. The `.git` location is *mostly* handled by `scm`, but we need to be careful when using `git` for hashing. And of course the actual working directory is implied whenever we forget to use an absolute path. `cwd` is therefore likely the worst name for the field, and is an artifact from the fact that the flag name is `--cwd`, with the assumption that the working directory _should_ be the directory that contains `turbo.json`. 
 
I'm going to suggest:
 * `cwd` -> `turboRoot` across the codebase, with a comment on `Config` explaining what it is (directory containing `turbo.json`)
 * do it in a separate PR